evalml/pipelines/pipeline_base.py

```diff
@@ -757,12 +757,22 @@ def _supports_fast_permutation_importance(self):
             for c in self.component_graph
         )
         has_dfs = any(isinstance(c, DFSTransformer) for c in self.component_graph)
+        if has_dfs:
+            dfs_feature_names = [
+                name
+                for feature in self.component_graph[DFSTransformer.name].features
```

Comment:
I 100% get why you didn't pull this out into a shared util with the `_handle_partial_dependence_fast_mode` method in featuretools.py (it's not exactly a lot of code), but I want to make sure that if we ever have to change one of these places for some reason that the other gets the same change made (I could see us forgetting they both exist!). Can we either have a comment linking the two places or pull out the shared logic into a util? It could even be a method on the DFS Transformer that just returns the boolean `contains_pre_existing_dfs_features`

Reply:
Done